include/pros/adi.hpp

```diff
@@ -23,6 +23,14 @@
 #include <cstdint>
 
 namespace pros {
+
+/** type definition for the pair of smart port and adi port for the basic adi devices */
+using ext_adi_port_pair_t = std::pair<std::uint8_t, std::uint8_t>;
```

Comment:
Pair and tuple are never included, which means the only way this compiled was because some other headers pulled them before adi.hpp got included.